webapp/components/interop.js

```diff
@@ -258,39 +260,108 @@ class WPTInterop extends QueryBuilder(
       this.handleSearchAutocomplete(e.detail.path);
     };
     this.onLoadingComplete = () => {
-      // passRateMetadata contains the url for the JSON blob of allMetrics;
+      // passRateMetadata contains the url for the JSON blob of precomputedInterop;
       // both fetches need to succeed + parse.
-      this.interopLoadFailed = !(this.passRateMetadata && this.allMetrics);
-      if (!this.interopLoadFailed && this.search) {
-        this.handleSearchCommit(this.search);
-      }
+      this.interopLoadFailed =
+        !(this.searchResults && this.searchResults.results && this.searchResults.results.length);
     };
   }
 
   connectedCallback() {
     super.connectedCallback();
-    this.shadowRoot.querySelector('test-search')
-      .addEventListener('commit', this.onSearchCommit);
-    this.shadowRoot.querySelector('test-search')
-      .addEventListener('autocomplete', this.onSearchAutocomplete);
+    this.testSearch.addEventListener('commit', this.onSearchCommit);
+    this.testSearch.addEventListener('autocomplete', this.onSearchAutocomplete);
   }
 
   disconnectedCallback() {
-    this.shadowRoot.querySelector('test-search')
-      .removeEventListener('commit', this.onSearchCommit);
+    this.testSearch.removeEventListener('commit', this.onSearchCommit);
     super.disconnectedCallback();
   }
 
   async ready() {
     await super.ready();
+    this._createMethodObserver('precomputedInteropLoaded(precomputedInterop)');
+    if (this.structuredQueries && this.searchCacheInterop) {
+      this.fetchSearchCacheInterop();
+    } else {
+      this.fetchPrecomputedInterop();
+    }
+  }
+
+  get testSearch() {
+    return this.shadowRoot.querySelector('test-search');
+  }
+
+  fetchPrecomputedInterop() {
     this.load(
       fetch(`/api/interop${this.query}`)
         .then(async r => {
           if (!r.ok || r.status !== 200) {
             Promise.reject('Failed to fetch interop data');
           }
-          this.passRateMetadata = await r.json();
-          this.allMetrics = await fetch(this.passRateMetadata.url).then(r => r.json());
+          const metadata = await r.json();
+          this.passRateMetadata = metadata;
+          this.testRuns = metadata && metadata.test_runs;
+          this.precomputedInterop = await fetch(this.passRateMetadata.url).then(r => r.json());
+        })
+        .then(() => {
+          if (this.search) {
+            this.handleSearchCommit(this.search);
+          }
+        })
+    );
+  }
+
+  fetchSearchCacheInterop() {
+    this.load(
+      Promise.resolve(this.testRuns || this.loadRuns())
+        .then(runs => {
+          if (!runs) {
```

Comment:
Looks like this is the only thing `runs` is used for? Can you either use `runs` below as well or `this.testRuns` in both places so that it's clear they're the same?

Reply:
Done.